src/ray/common/scheduling/cluster_resource_scheduler.cc

```diff
@@ -885,7 +886,9 @@ bool ClusterResourceScheduler::AllocateResourceInstances(
 }
 
 bool ClusterResourceScheduler::AllocateTaskResourceInstances(
-    const TaskRequest &task_req, TaskResourceInstances *task_allocation /* return */) {
+    const TaskRequest &task_req, 
+    std::shared_ptr<TaskResourceInstances> task_allocation /* return */) {
```

Comment:
By convention this should be a pointer to a `shared_ptr`. Then you can remove the `return` comment because it's clear.